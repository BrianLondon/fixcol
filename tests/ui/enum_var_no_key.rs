use fixed_derive::ReadFixed;

#[derive(ReadFixed)]
struct MyStruct {
    #[fixed(width = 5)]
    my_int: u16
}

#[derive(ReadFixed)]
#[fixed(key_width = 3)]
enum Thing1 {
    #[fixed(key = "one")]
    Thing1(#[fixed(width = 5)] u16),
    #[fixed(embed = true)]
    Thing2(MyStruct),
}

#[derive(ReadFixed)]
#[fixed(key_width = 3)]
enum Thing2 {
    Thing1(#[fixed(width = 5)] u16),
    #[fixed(embed = true)]
    Thing2(MyStruct),
}

pub fn main() {}
