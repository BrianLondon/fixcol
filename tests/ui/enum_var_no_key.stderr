error: fixed-derive error: The parameter key must be provided for all enum variants.

        Try adding #[fixed(key = "<my key>")] to this variant.
  --> tests/ui/enum_var_no_key.rs:15:5
   |
15 |     Thing2(MyStruct),
   |     ^^^^^^

error: fixed-derive error: The parameter key must be provided for all enum variants.

        Try adding #[fixed(key = "<my key>")] to this variant.
  --> tests/ui/enum_var_no_key.rs:21:5
   |
21 |     Thing1(#[fixed(width = 5)] u16),
   |     ^^^^^^
